src/DocumentFormat.OpenXml/SimpleTypes/OpenXmlComparableReferenceType.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace DocumentFormat.OpenXml
+{
+    /// <summary>
+    /// Represents a comparable and equatable reference.
+    /// </summary>
+    /// <typeparam name="T">The type of the value.</typeparam>
+    public abstract class OpenXmlComparableReferenceType<T> : OpenXmlSimpleType,
+        IComparable, IComparable<OpenXmlComparableReferenceType<T>>, IEquatable<OpenXmlComparableReferenceType<T>>
+        where T : class, IComparable, IComparable<T>, IEquatable<T>
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="OpenXmlComparableReferenceType{T}"/>.
+        /// </summary>
+        private protected OpenXmlComparableReferenceType()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OpenXmlComparableReferenceType{T}"/>
+        /// class by deep copying the supplied <see cref="OpenXmlComparableReferenceType{T}"/>
+        /// value.
+        /// </summary>
+        /// <param name="source">The source <see cref="OpenXmlComparableReferenceType{T}"/> instance.</param>
+        private protected OpenXmlComparableReferenceType(OpenXmlComparableReferenceType<T> source) : base(source)
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the value of this instance.
+        /// </summary>
+        public abstract T Value { get; set; }
+
+        /// <inheritdoc />
+        public int CompareTo(object obj)
+        {
+            if (obj is OpenXmlComparableReferenceType<T> other)
+            {
+                return CompareTo(other);
+            }
+
+            throw new ArgumentException();
```

Comment:
Should there be a message here?

Reply:
Easy answer: yes.

Reply:
haha now the hard question is figuring out what to put :)

Reply:
Here's my current thinking (and I am very much open for suggestions):
```
        public int CompareTo(object obj)
        {
            switch (obj)
            {
                case null:
                    return 1;

                case OpenXmlComparableReferenceType<T> other:
                    return CompareTo(other);
            }

            throw new ArgumentException("Incompatible type: " + obj.GetType().FullName);
        }
```
I also used a switch statement with pattern matching.

Reply:
I'm fine with that - just stick it in the resource file